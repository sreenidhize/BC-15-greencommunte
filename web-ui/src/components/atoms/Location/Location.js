import React from "react";
import PropTypes from "prop-types";
import RoomOutlinedIcon from "@material-ui/icons/RoomOutlined";
import { makeStyles } from "@material-ui/core";
import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles((theme) => ({
  root: {
    color: theme.palette.secondary.main,
    width: "20px",
    height: "20px",
  },
  label: {
    fontSize: "16px",
    paddingLeft: "10px",
    paddingTop: "3px",
    display: "inline",
    position: "absolute",
  },
}));

export const Location = ({ label }) => {
  const classes = useStyles();

  return (
    <div data-testid="location-placeholder">
      <RoomOutlinedIcon className={classes.root}></RoomOutlinedIcon>
      <Typography className={classes.label} variant="h4">
        {label}
      </Typography>
    </div>
  );
};

Location.propTypes = {
  label: PropTypes.string,
  icon: PropTypes.string,
};

Location.defaultProps = {
  label: "label",
};
