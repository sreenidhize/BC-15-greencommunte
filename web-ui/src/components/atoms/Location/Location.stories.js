import React from "react";
import { Location } from "./Location";

export default {
  title: "atoms/Location",
  component: Location,
};

const Template = (args) => <Location {...args} />;

export const Secundrabad = Template.bind({});
Secundrabad.args = {
  label: "East Marredpally, Secundrabad",
};
