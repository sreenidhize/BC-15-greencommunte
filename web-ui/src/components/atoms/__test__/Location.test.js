import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Location } from '../Location/Location';

describe("Location placeholder test", () => {
    it("Matches the snapshot", () => {
      const { container } = render(
        <Location label="Hyderabad" />
      );
      expect(container).toMatchSnapshot();
    });
  
    test("test Location render", () => {
      const { getByTestId } = render(<Location label="Hyderabad" />);
      expect(getByTestId("location-placeholder")).toBeInTheDocument();
    });
  });