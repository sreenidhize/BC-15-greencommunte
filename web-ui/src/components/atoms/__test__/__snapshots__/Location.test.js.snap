// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location placeholder test Matches the snapshot 1`] = `
<div>
  <div
    data-testid="location-placeholder"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-root-1"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zM7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 2.88-2.88 7.19-5 9.88C9.92 16.21 7 11.85 7 9z"
      />
      <circle
        cx="12"
        cy="9"
        r="2.5"
      />
    </svg>
    <h4
      class="MuiTypography-root makeStyles-label-2 MuiTypography-h4"
    >
      Hyderabad
    </h4>
  </div>
</div>
`;
